### PR TITLE
JAMES-3386 create mailbox with blank path should not be allowed

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java [hunk 1]
```diff
@@ -194,10 +195,11 @@ private boolean isMailboxNameTooLong() {
 
     boolean hasEmptyNameInHierarchy(char pathDelimiter) {
         String delimiterString = String.valueOf(pathDelimiter);
-        return this.name.isEmpty()
-            || this.name.contains(delimiterString + delimiterString)
-            || this.name.startsWith(delimiterString)
-            || this.name.endsWith(delimiterString);
+        String nameWithoutSpaces = name.replaceAll("\\s", "");
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java:198]: You need unit tests in MailboxPathTest too

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test thos please

**Comment by chibenwa** [comment]
test this please

File: mailbox/api/src/test/java/org/apache/james/mailbox/model/MailboxPathTest.java [hunk 2]
```diff
@@ -220,6 +248,12 @@ void hasEmptyNameInHierarchyShouldBeTrueIfPathWithAnEmptyNameBetweenTwoNames() {
             .hasEmptyNameInHierarchy('.'))
             .isTrue();
     }
+    @Test
+    void hasEmptyNameInHierarchyShouldBeTrueIfPathWithAaBlankNameBetweenTwoNames() {
```
Comment (Arsnael) [inline mailbox/api/src/test/java/org/apache/james/mailbox/model/MailboxPathTest.java:252]:
```suggestion
    void hasEmptyNameInHierarchyShouldBeTrueIfPathWithABlankNameBetweenTwoNames() {
```
Also skip a line before `@Test` tag

**Comment by chibenwa** [comment]
```
[ERROR] Failed to execute goal on project james-server-util: Could not resolve dependencies for project org.apache.james:james-server-util:jar:3.6.0-SNAPSHOT: Failed to collect dependencies at com.github.steveash.guavate:guavate:jar:1.0.0 -> com.google.guava:guava:jar:25.1-jre -> com.google.errorprone:error_prone_annotations:jar:2.1.3: Failed to read artifact descriptor for com.google.errorprone:error_prone_annotations:jar:2.1.3: Could not transfer artifact com.google.errorprone:error_prone_parent:pom:2.1.3 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/com/google/errorprone/error_prone_parent/2.1.3/error_prone_parent-2.1.3.pom.part (No such file or directory) -> [Help 1]
```

Bad luck

test this please

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 3]
```diff
@@ -657,6 +711,119 @@ trait MailboxSetMethodContract {
...
+        |}
+        |""".stripMargin
+
+    val response: String =
+      `given`
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(request)
+        .when
+        .post
+        .`then`
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:743]: indent

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 4]
```diff
@@ -657,6 +711,119 @@ trait MailboxSetMethodContract {
...
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:799]: indent

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by nvduc91 — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please 

Deployment test

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**